plugins/LadspaEffect/swh/CMakeLists.txt

```diff
@@ -16,6 +16,14 @@ SET(COMPILE_FLAGS "${COMPILE_FLAGS} ${PIC_FLAGS}")
 # Loop over every XML file
 FILE(GLOB XML_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/ladspa/*.xml")
 LIST(SORT XML_SOURCES)
+
+IF(LMMS_BUILD_APPLE)
+	# Prefer system perl over Homebrew, MacPorts, etc
+	SET(PERL_COMMAND "/usr/bin/perl")
+ELSE()
+	SET(PERL_COMMAND "perl")
+ENDIF()
```

Comment:
@tresf Can't we just `find_package(Perl REQUIRED)` on non-macOS platforms and use `PERL_EXECUTABLE` instead? CMake 3.12 or later also allows searching for `/usr/bin/perl` first by setting `Perl_ROOT` to `/usr/bin` on macOS.

Reply:
I think `find_package(...)` is always preferred.  `Perl_ROOT` is a nice workaround as well.

Reply:
> @tresf Can't we just `find_package(Perl REQUIRED)` on non-macOS platforms and use `PERL_EXECUTABLE` instead? CMake 3.12 or later also allows searching for `/usr/bin/perl` first by setting `Perl_ROOT` to `/usr/bin` on macOS.

So this will add the following changes to the project:

```cmake
+ CMAKE_MINIMUM_REQUIRED(VERSION 3.12)
+ CMAKE_POLICY(SET CMP0074 NEW)
```

It will also move the Perl detection to the root `CMakeLists.txt` so that `swh` can be flagged on and off appropriately (rather than `REQUIRED`).

Preparing a commit now.  Feel free to patch it if the build fails for trivial reasons.

Reply:
Can we have that one as a seperate pr?

Reply:
Done: 12a1300

Reply:
> Can we have that one as a seperate pr?

I think this is the place for it, since MSVC support impacts Perl detection. 🍻

Reply:
It seems like we're still using Ubuntu 18.04 in CI which only provides CMake 3.10.

Reply:
@PhysSong if you can, please fix it or revert it. I think perl detection is the way to go, but the policy needs to be conditional now.

Reply:
Then I'd say revert those changes for now. We can do that in a separate PR once we upgrade all CI builds to use 20.04 or later.

Reply:
> It seems like we're still using Ubuntu 18.04 in CI which only provides CMake 3.10.

Is it possible to bump to ubuntu 20.04?

Reply:
I believe I tried this on WSL and it worked just fine using repository pinning.

Reply:
On second thought, this is is probably a bad idea because it would make the mingw steps a bit harder.  I'll see if I can add a condition with a FIXME note.

Reply:
> Then I'd say revert those changes for now. We can do that in a separate PR once we upgrade all CI builds to use 20.04 or later.

@PhysSong I hope you agree, making this conditional satisfies the PR best:
1. Makes perl a dependency -- because it is
2. If Mac and newer CMake is detected, enables the `Perl_ROOT` variable

Done via 329613b.

This can have some unintended side-effects:
1. Future use of the `<Package>_ROOT` will have be ignored in Ubuntu 18.04 environments, may be confusing to developers
2. Older Mac environments will likely fail

I think the benefit of having Perl as a dependency is a good trade-off.  This way other platforms have more usable error messages when building (e.g. Haiku, etc).

The builds are passing now.  If everyone's OK with this, we can resolve this conversation.

Reply:
@tresf Sounds good. Resolving.